src/persist/src/indexed/mod.rs

```diff
@@ -835,12 +832,16 @@ impl<L: Log, B: Blob> Indexed<L, B> {
 
         for (id, updates) in updates {
             if let Some(listen_fns) = self.listeners.get(&id) {
-                if listen_fns.len() == 1 {
-                    listen_fns[0].0(ListenEvent::Records(updates));
-                } else {
-                    for listen_fn in listen_fns.iter() {
-                        listen_fn.0(ListenEvent::Records(updates.clone()));
-                    }
+                for listen_fn in listen_fns.iter() {
+                    let records = ListenEvent::Records(
```

Comment:
let's do this once and clone it instead of once per listener (and then keep the optimization from before where we don't clone if there's only one listener), making sure to skip that work if there's 0 listeners